promslog/slog_test.go

```diff
@@ -43,29 +43,29 @@ func TestDefaultConfig(t *testing.T) {
 }
 
 func TestUnmarshallLevel(t *testing.T) {
-	l := &AllowedLevel{}
+	l := NewLevel()
 	err := yaml.Unmarshal([]byte(`debug`), l)
 	if err != nil {
 		t.Error(err)
 	}
-	if l.s != "debug" {
-		t.Errorf("expected %s, got %s", "debug", l.s)
+	if got := l.String(); got != "debug" {
+		t.Errorf("expected %s, got %s", "debug", got)
 	}
 }
 
 func TestUnmarshallEmptyLevel(t *testing.T) {
-	l := &AllowedLevel{}
+	l := NewLevel()
 	err := yaml.Unmarshal([]byte(``), l)
 	if err != nil {
 		t.Error(err)
 	}
-	if l.s != "" {
-		t.Errorf("expected empty level, got %s", l.s)
+	if got := l.String(); got != "info" {
```

Comment:
I changed the behaviour purposefully. This simplifies (and optimizes) the level code.

I don't know exactly how Unmarshal and String was envisioned to be used in the YAML context. Does this blocks anyone?